docusaurus/docs/create-a-plugin/develop-a-plugin/good-practices.md

```diff
@@ -0,0 +1,66 @@
+---
+id: good-practices
+title: Good practices
+description: An index of good practices for plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - good practices
+sidebar_position: 5
+---
+
+This document summarizes good practices for Grafana plugins. This live document is updated as new good practices are identified.
+
+## General
+
+- Choose the right plugin type for your use case: panel, data-source or app. See [Plugin types](https://grafana.com/developers/plugin-tools/introduction/plugin-types-usage) for more information.
+- Verify that your data-source or app plugin can be provisioned. See [Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- Include default dashboards in your data-source or app plugin. See [Bundling of dashboards](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- Ensure that the minimum version of Grafana is correct. This should point to the earliest version of Grafana that your plugin fully supports. This is specified as `grafanaDependency` in your plugin.json.
```

Comment:
Worth mentioning this semver should align with the version of the npm packages the plugin is developed with?

Reply:
You mean that the minimum `grafanaDependency` should match the `@grafana/*` packages version? If that's the case, I am not sure, some people may use some automation to upgrade those packages, which is not something wrong?